go/fn/object.go

```diff
@@ -759,6 +784,107 @@ func (o KubeObjects) Less(i, j int) bool {
 	return idStrI < idStrJ
 }
 
+func (o KubeObjects) String() string {
+	var elems []string
+	for _, obj := range o {
+		elems = append(elems, strings.TrimSpace(obj.String()))
+	}
+	return strings.Join(elems, "\n---\n")
+}
+
+// Select will return the subset of objects in KubeObjects such that f(object) returns 'true'.
+func (o KubeObjects) Select(f func(o *KubeObject) bool) KubeObjects {
+	var result KubeObjects
+	for _, obj := range o {
+		if f(obj) {
+			result = append(result, obj)
+		}
+	}
+	return result
+}
+
+// SelectByGvk will return the subset of objects that matches the provided GVK.
+func (o KubeObjects) SelectByGvk(apiVersion, kind string) KubeObjects {
+	return o.Select(func(o *KubeObject) bool {
+		return o.IsGVK(apiVersion, kind)
```

Comment:
Can we expose (or have we already exposed) `func IsGVK(apiVersion, kind) func (o *KubeObject) bool {...` so we can do `o.Select(IsGVK("v1", "ConfigMap"))`?

I think including that function will simplify the implementation here.  I don't know if we then need `SelectByGVK` and friends, I have no real objection to them, but it may be that we can add them if/when `Select(IsGVK(...))` gets annoying?

Reply:
> `func IsGVK(apiVersion, kind) func (o *KubeObject) bool {`

It took me a second to wrap my head around what this syntax means, and I didn't realize you could do this. But I really like this, and will replace `SelectByGvk` and friends with this!